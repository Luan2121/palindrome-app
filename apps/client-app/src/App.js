import AppProviders from './AppProviders';

// components
import Navbar from "./components/Navbar";
import Results from "./components/Results";
import './bootstrap';

function App() {
  return (
    <AppProviders>
      <Navbar />
      <Results />
    </AppProviders>
  )
}

export default App;
