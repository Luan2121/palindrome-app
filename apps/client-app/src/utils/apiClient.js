const { REACT_APP_BASE_API_URL : baseApiUrl = 'http://localhost:4000' } = process.env;

const client = (
    url,
    { method = 'GET', data, headers : customHeaders, ...customConfig }
) => {
    const endpoint = `${baseApiUrl}/${url}`;
    const config = {
        method,
        headers: {
            'Content-Type':  data ? 'application/json' : undefined,
            'Accept': 'application/json',
            ...customHeaders
        },
        ...customConfig
    };
    return fetch(endpoint,config)
        .then( res => res.json() )
}

export { client };