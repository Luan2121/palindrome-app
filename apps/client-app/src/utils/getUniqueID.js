

const getUniqueID = () => {
    return Date.now()
}

export { getUniqueID }