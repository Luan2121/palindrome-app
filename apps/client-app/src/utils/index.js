export * from './apiClient';
export * from './checkPalindrome';
export * from './getUniqueID';