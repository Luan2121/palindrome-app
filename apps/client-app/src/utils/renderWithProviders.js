import { render } from "@testing-library/react";
import AppProviders from "../AppProviders";

export const renderWithProviders = (component) => {
    return render(
        <AppProviders>
            {component}
        </AppProviders>
    )
}