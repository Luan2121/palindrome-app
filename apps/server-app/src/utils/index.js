const reverseString = str => {
  if (typeof str !== 'string') {
    return ''
  }
  return str.split('').reverse().join('')
}

const parseText = str => {
  if (typeof str !== 'string') {
    return ''
  }

  const nonSpecialCharactersRegex = /[^a-zA-Z0-9]/g
  return str
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '')
    .toLowerCase()
    .replace(nonSpecialCharactersRegex, '')
}

const isPalindrome = text => {
  const parsedText = parseText(text)
  const reversedText = reverseString(parsedText)
  return reversedText === parsedText
}

export { reverseString, isPalindrome }
