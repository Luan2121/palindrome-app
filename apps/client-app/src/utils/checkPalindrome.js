import { client } from './apiClient';

const checkPalindrome = (text) => {
    const params = { text };
    return client(`iecho?${ new URLSearchParams(params).toString()}`,{});
}

export { checkPalindrome }