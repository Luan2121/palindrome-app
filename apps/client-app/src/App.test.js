import App from './App';
import { renderWithProviders } from './utils/renderWithProviders';

describe('App Component', () => {

    test('it should render without crashing', () => {
        const { asFragment } = renderWithProviders(<App />);
        expect( asFragment() ).toMatchSnapshot();
    })

})

