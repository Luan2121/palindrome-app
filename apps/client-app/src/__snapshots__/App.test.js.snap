// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component it should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="bg-secondary p-4 d-flex justify-content-center m-0 container-fluid"
  >
    <div
      class="justify-content-center w-100 row row-cols-md-6"
    >
      <form
        class="d-flex col-md-6"
      >
        <div
          class="me-3 w-100"
        >
          <input
            class="form-control"
            placeholder="Insert Text"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          disabled=""
          type="submit"
        >
          Send
        </button>
      </form>
    </div>
  </div>
  <main
    class="p-5 container"
  >
    <div
      class="p-4 bg-body w-100"
    >
      <h2
        class="mb-4 d-block"
      >
        Results:
      </h2>
      <div
        class="justify-content-center row"
      >
        <div
          class="text-center"
        >
          <h3>
            It looks like you haven't used the app
          </h3>
          <p>
            Try it wrinting a text like: Was it a car or a cat I saw.
          </p>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
